portal-impl/src/com/liferay/portal/util/HttpImpl.java

```diff
@@ -1444,6 +1449,69 @@ protected CloseableHttpClient getCloseableHttpClient(HttpHost proxyHost) {
 		return _closeableHttpClient;
 	}
 
+	protected int getMaxURLDepth(
+		String url, int curLength, int maxLength, int count) {
+
+		String ampersand = StringPool.AMPERSAND;
+		String equal = StringPool.EQUAL;
+		String question = StringPool.QUESTION;
+
+		for (int i = 0; i < count; i++) {
```

Comment:
This method is recursive, so with each level of encoding, the delimiters and characters which are encodable must also be encoded to make sure we are searching at the correct depth